### ISSUE TITLE
No results when screenshots in Dropbox folder

### ISSUE THREAD

**Issue by gabrielwhite** [vitorgalvao/screenshots-workflow#3]
### Workflow version

2024.4

### Alfred version

5.5

### macOS version

14.5

### Debugger output

```alfred_debugger
[09:14:35.171] Screenshots[Keyword] Processing complete
[09:14:35.179] Screenshots[Keyword] Passing output '' to Run Script
[09:14:35.256] Screenshots[Run Script] Processing complete
[09:14:35.261] Screenshots[Run Script] Passing output '{"items":[{"title":"No Results",
"subtitle":"Did not find any screenshots",
"valid":false}]}
' to Grid View
```

### More details

My screenshot folder is set to: 

~/Library/CloudStorage/Dropbox/Screenshots

I get no results when I invoke the workflow.

I can get results when I move the same screenshots to my Desktop and set the Desktop as my search folder.

Thank you!

**Comment by gabrielwhite** [comment]
For what it's worth here's the debugger log from the Desktop:


```
[09:16:23.574] Screenshots[Keyword] Processing complete
[09:16:23.583] Screenshots[Keyword] Passing output '' to Run Script
[09:16:23.659] Screenshots[Run Script] Processing complete
[09:16:23.663] Screenshots[Run Script] Passing output '{"items":[{"uid":"\/Users\/gabe\/Desktop\/Screenshot 2024-05-16 at 10.18.12 AM.png","title":"Screenshot 2024-05-16 at 10.18.12 AM.png","icon":{"path":"\/Users\/gabe\/Desktop\/Screenshot 2024-05-16 at 10.18.12 AM.png"},"type":"file","arg":"\/Users\/gabe\/Desktop\/Screenshot 2024-05-16 at 10.18.12 AM.png","subtitle":"~\/Desktop\/Screenshot 2024-05-16 at 10.18.12 AM.png"},{"icon":{"path":"\/Users\/gabe\/Desktop\/Screenshot 2024-05-16 at 10.16.30 AM.png"},"arg":"\/Users\/gabe\/Desktop\/Screenshot 2024-05-16 at 10.16.30 AM.png","title":"Screenshot 2024-05-16 at 10.16.30 AM.png","subtitle":"~\/Desktop\/Screenshot 2024-05-16 at 10.16.30 AM.png","uid":"\/Users\/gabe\/Desktop\/Screenshot 2024-05-16 at 10.16.30 AM.png","type":"file"},{"subtitle":"~\/Desktop\/Screenshot 2024-05-15 at 3.50.33 PM.png","type":"file","icon":{"path":"\/Users\/gabe\/Desktop\/Screenshot 2024-05-15 at 3.50.33 PM.png"},"title":"Screenshot 2024-05-15 at 3.50.33 PM.png","uid":"\/Users\/gabe\/Desktop\/Screenshot 2024-05-15 at 3.50.33 PM.png","arg":"\/Users\/gabe\/Desktop\/Screenshot 2024-05-15 at 3.50.33 PM.png"}]}
' to Grid View

```

**Comment by vitorgalvao** [comment]
Open a terminal and run `mdls '/path/to/screenshot'`. One of the entries you’ll see is `kMDItemIsScreenCapture`. The workflow works by querying Spotlight’s database for files where that value is `1`. If the files in your Dropbox don’t have that or aren’t indexed by Spotlight, they won’t show up.

Because it’s working when you move the file to your Desktop, a likely explanation is that Dropbox is uploading those files and removing the local copy. Then when you move it to the Desktop, they have to download the file again. If that’s the case, you have to ensure that folder is always available offline. The other possibility is that folder isn’t being indexed at all, in which case you should try [the troubleshooting steps](https://www.alfredapp.com/help/troubleshooting/indexing/).

Whatever the cause, this will be down to the interaction between Spotlight and Dropbox. Fix that and the workflow will work for you.

**Comment by gabrielwhite** [comment]
Thanks for the quick response.

I verified that the images are tagged as screenshots (`kMDItemIsScreenCapture=1`), and I verified that the screenshots are appearing in Spotlight search results. I also verified that the Dropbox folder is an offline folder. 

I also forced a rebuild of the Spotlight index ([this approach](https://www.dropboxforum.com/t5/Create-upload-and-share/Dropbox-is-not-searchable-in-Mac-OS-Monterey-12-0-1/m-p/626720/highlight/true#M58591)).

The results are still not appearing in the workflow. 

I'll probably just move my screenshots folder out of Dropbox - it's not critical that it's in there, but it would be nice to have a fix for this.

Thanks again!
